engine/src/main/java/org/terasology/rendering/dag/nodes/RefractiveReflectiveBlocksNode.java

```diff
@@ -135,7 +140,7 @@ public void process() {
 
     @Override
     public void update() {
-        // TODO: renames, maybe?
-        READ_ONLY_GBUFFER.attachDepthBufferTo(displayResolutionDependentFBOs.get(REFRACTIVE_REFLECTIVE));
+        // TODO: Review this code, it seems redundant because the same code is called in process() of ApplyDeferredLightingNode
```

Comment:
Ha! Thank you for mentioning this. I had a quick look and it looks like this line is probably necessary... but for some intricate reason! 

Whenever the read/write gBuffers are swapped the refl-refr FBO apparently needs to be reconfigured to share the depth buffer with one of the gBuffers. And imagine the same requirement applies when the display is resized.

A naively logical place where to do this is after the swap takes place, as it's done in ApplyDeferredLightingNode. But this requires that node to have knowledge of something otherwise unnecessary to it. Another place where it _should_ happen according to this reasoning is the PrePostCompositeNode, as it swaps the buffer. But in that node the the attachDepthBufferTo line is commented out!

This all seems to make a complicated picture, but I suspect the solution is quite a simple one: the only attachDepthBufferTo line that is actually necessary is this one as long as the update method gets called when the FBOs are regenerated and swapped. Which I think is the case because this node subscribes to the fbo manager. 

So, the following line is most likely the correct one, the TODO line above can be removed but also the attachDepthBufferTo lines in ApplyDeferredLightingNode and PrePostCompositeNode can be removed.